### PR TITLE
feat: new ui kit components

### PR DESCRIPTION
**Pull request by kandl**
Add guide to write new components.

Add new components to the UI kit:
- UiButton
- UiIcon

Also add screenshot tests for them.

Add default theme and theming validation, to avoid growing
of the inconsistent default values in the theming variables.

Add generation of the default scss theme.

Suffix theming variables with inconsistent default values with
`-from-theme` suffix, so we can start to use the default theme
in scss and still keep theming backward compatible.
These variables will be set only if ThemeProvider will set them,
so they will be themed the same way as before, but keep
fall backing to the default value, even with default theme set.

Add experimental ScopedThemeProvider.

risk: low
JIRA: F1-123

---

> [!IMPORTANT]
> Please, **don't forget to run `rush change`** for the commits that introduce **new features** 🙏

---

Refer to [documentation](https://github.com/gooddata/gooddata-ui-sdk/blob/master/dev_docs/continuous_integration.md) to see how to run checks and tests in the pull request. This is the list of the most used commands:

```
extended test - backstop
```

```
extended test - tiger-cypress - integrated
extended test - tiger-cypress - isolated
extended test - tiger-cypress - record
```

### CODE REVIEW

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2718/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by kandl** [comment]
extended test - backstop

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2719/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by kandl** [comment]
extended test - backstop

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2721/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2720/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by kandl** [comment]
extended test - backstop

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2722/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2723/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

File: libs/sdk-ui-kit/README-DEV.MD [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+Interface should be aligned with Figma as much as possible - design of the interface should be discussed together with respective UX and UI personas.
+
+## Styles
+
+Styles should follow the [BEM](https://en.bem.info/methodology/quick-start/) methodology.
+
+Follow these rules:
+- Use prefixes for root elements to avoid class name collisions (e.g., `.gd-ui-kit-button`, `.gd-ui-kit-icon`).
+- Minimize nesting and overrides for better readability and maintainability
+- Avoid using margins for root elements; components should not affect the layout of their parent or sibling elements.
```
Comment (xMort) [inline libs/sdk-ui-kit/README-DEV.MD:35]: "must not" instead of "should not" in this and the next sentence.

File: libs/sdk-ui-kit/README-DEV.MD [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+
+All default theme css properties values can be found in [defaultTheme.scss](../sdk-ui-kit/src/@ui/defaultTheme.scss).
+
+If you are adding a new CSS property, you should add its specification [here](../sdk-ui-theme-provider/src/variablesSpec/allThemeCssVariables.ts), to its respective location.
+This is required, because these specifications are used to generate default theme, and for validation of the consistent properties usage to keep our theming consistent.
+
+
+## Layouting
+
+Component should care only about its own layout.
```
Comment (xMort) [inline libs/sdk-ui-kit/README-DEV.MD:254]: Must here and the next sentence.

File: libs/sdk-ui-kit/README-DEV.MD (hunk 1 above)
Comment (kandl) [reply libs/sdk-ui-kit/README-DEV.MD]: Updated. I also added new readme for more details about theming (see the last commit).

File: libs/sdk-ui-kit/README-DEV.MD (hunk 2 above)
Comment (kandl) [reply libs/sdk-ui-kit/README-DEV.MD:252]: Updated. I also added new readme for more details about theming (see the last commit).

**Comment by kandl** [comment]
extended test - backstop

**Review by xMort — APPROVED** [review]

**Review by xMort — APPROVED** [review]

**Comment by kandl** [comment]
extended test - backstop

**Comment by kandl** [comment]
extended test - backstop

**Comment by kandl** [comment]
extended test - backstop

File: libs/sdk-ui-theme-provider/src/ThemeProvider/ScopedThemeProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+import identity from "lodash/identity.js";
+import { useBackend, useWorkspace } from "@gooddata/sdk-ui";
+import { IAnalyticalBackend } from "@gooddata/sdk-backend-spi";
+import { ITheme } from "@gooddata/sdk-model";
+
+import { clearCssProperties, setCssProperties } from "../cssProperties.js";
+import {
+    ThemeContextProvider,
+    ThemeStatus,
+    // useTheme
```
Comment (ivanmjartan) [inline libs/sdk-ui-theme-provider/src/ThemeProvider/ScopedThemeProvider.tsx:12]: Remove comment

**Review by ivanmjartan — APPROVED** [review]

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2724/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by kandl** [comment]
extended test - backstop

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2725/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Review by xMort — APPROVED** [review]

**Pull request merged**